bears/vcs/git/GitCommitBear.py

```diff
@@ -151,6 +162,59 @@ def run(self,
             body,
             **self.get_issue_checks_metadata().filter_parameters(kwargs))
 
+    def check_revert(self, shortlog, body,
+                     allow_revert_commits_without_reason: bool = False):
+        """
+        Validates revert commits.
+
+        :param shortlog:
+            The shortlog message string.
+        :param body:
+            The body of the commit message of HEAD.
+        :param allow_revert_commits_without_reason:
+            Whether reason is needed for revert commits.
+        """
+        if not shortlog.startswith('Revert '):
+            return
+
+        shortlog_start_index = shortlog.find('"')
+        shortlog_end_index = shortlog.rfind('"')
+        reverted_shortlog = shortlog[shortlog_start_index +
+                                     1: shortlog_end_index]
+
+        body = body.strip('\n')
+        blankline = body.find('\n\n')
+        headline = body[:blankline] if blankline != -1 else body
+
+        if not headline.startswith('This reverts '):
+            yield Result(self, 'Invalid revert commit.')
+            return
+
+        sha_regex = re.compile(r'([0-9a-f]{40})')
```

Comment:
As the intent is also to maintain readability why not support short SHA1 hashes?

Reply:
`git revert` doesnt do that